homeassistant/components/coolmaster/__init__.py

```diff
@@ -17,4 +38,9 @@ async def async_setup_entry(hass, entry):
 
 async def async_unload_entry(hass, entry):
     """Unload a Coolmaster config entry."""
-    return await hass.config_entries.async_forward_entry_unload(entry, "climate")
+    unload_ok = await hass.config_entries.async_forward_entry_unload(entry, "climate")
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
```

Comment:
```suggestion
    return unload_ok


class CoolmasterDataUpdateCoordinator(DataUpdateCoordinator):
    """Class to manage fetching Coolmaster data."""

    def __init__(self, hass, *, coolmaster):
        """Initialize global Coolmaster data updater."""
        self.coolmaster = coolmaster
        update_interval = timedelta(seconds=60)

        super().__init__(
            hass, _LOGGER, name=DOMAIN, update_interval=update_interval,
        )

    async def _async_update_data(self):
        """Fetch data from Coolmaster."""
        try:
            return await self.coolmaster.status()
        except Exception as err:
            raise UpdateFailed(f"Update failed: {err}")
 
```
```

Reply:
this will need more error checking depending on your pypi module

Reply:
What's the reason to use timedelta(seconds=60) instead of importing SCAN_INTERVAL from Climate? The current (non-async) implementation uses the SCAN_INTERVAL